meshio/gmsh/_gmsh41.py

```diff
@@ -62,12 +63,20 @@ def read_buffer(f, is_ascii, data_size):
         if environ == "PhysicalNames":
             _read_physical_names(f, field_data)
         elif environ == "Entities":
-            physical_tags = _read_entities(f, is_ascii, data_size)
+            # Read physical tags and information on bounding entities.
+            # The information is passed to the processing of elements.
+            physical_tags, bounding_entities = _read_entities(f, is_ascii, data_size)
         elif environ == "Nodes":
-            points, point_tags = _read_nodes(f, is_ascii, data_size)
+            points, point_tags, point_entities = _read_nodes(f, is_ascii, data_size)
```

Comment:
Question: What are `point_entities` in gmsh lingo?

Reply:
From what I understand, Gmsh keeps track of which entities (geometric objects in the .geo file) points and cells have been assigned to:
* in the $Nodes section, the nodes are grouped into these entities. Since the entity numbering starts from 1 in each dimension, the dimension of the entity of the node must be stored. 
* the $Elements section is similar, but it contains only entities for which cells were generated (e.g. only those decleared Physical in gmsh lingo). Thus the entity information for cells is a subset of the full entity set.

The $Entities section is (I believe) generated from the entities of nodes, with physical tags and boundary representation from the cells. I have no idea how all this is handled internally in gmsh, but when I tried to get away with preserving less information, gmsh could not process the file.